server/datastore/mysql/query_results_test.go

```diff
@@ -44,79 +43,125 @@ func testGetQueryResultRows(t *testing.T, ds *Datastore) {
 		},
 	}
 
-	err = ds.SaveQueryResultRows(context.Background(), resultRow3)
+	err = ds.SaveQueryResultRows(context.Background(), query2Rows)
 	require.NoError(t, err)
 
-	// Assert that Query1 returns 2 results
-	results, err := ds.QueryResultRowsForHost(context.Background(), resultRows[0].QueryID, resultRows[0].HostID)
+	results, err := ds.QueryResultRows(context.Background(), query.ID)
 	require.NoError(t, err)
-	require.Len(t, results, 2)
-	require.Equal(t, resultRows[0].QueryID, results[0].QueryID)
-	require.Equal(t, resultRows[0].HostID, results[0].HostID)
-	require.Equal(t, resultRows[0].LastFetched.Unix(), results[0].LastFetched.Unix())
-	require.JSONEq(t, string(resultRows[0].Data), string(results[0].Data))
-	require.Equal(t, resultRows[1].QueryID, results[1].QueryID)
-	require.Equal(t, resultRows[1].HostID, results[1].HostID)
-	require.Equal(t, resultRows[1].LastFetched.Unix(), results[1].LastFetched.Unix())
-	require.JSONEq(t, string(resultRows[1].Data), string(results[1].Data))
+	require.Len(t, results, 1) // Should not return rows with nil data
+	require.Equal(t, query1Rows[1].QueryID, results[0].QueryID)
+	require.Equal(t, query1Rows[1].HostID, results[0].HostID)
+	require.Equal(t, query1Rows[1].LastFetched.Unix(), results[0].LastFetched.Unix())
+	require.JSONEq(t, string(*query1Rows[1].Data), string(*results[0].Data))
 
 	// Assert that Query2 returns 1 result
-	results, err = ds.QueryResultRowsForHost(context.Background(), resultRow3[0].QueryID, resultRow3[0].HostID)
+	results, err = ds.QueryResultRows(context.Background(), query2.ID)
 	require.NoError(t, err)
 	require.Len(t, results, 1)
-	require.Equal(t, resultRow3[0].QueryID, results[0].QueryID)
-	require.Equal(t, resultRow3[0].HostID, results[0].HostID)
-	require.Equal(t, resultRow3[0].LastFetched.Unix(), results[0].LastFetched.Unix())
-	require.JSONEq(t, string(resultRow3[0].Data), string(results[0].Data))
+	require.Equal(t, query2Rows[0].QueryID, results[0].QueryID)
+	require.Equal(t, query2Rows[0].HostID, results[0].HostID)
+	require.Equal(t, query2Rows[0].LastFetched.Unix(), results[0].LastFetched.Unix())
+	require.JSONEq(t, string(*query2Rows[0].Data), string(*results[0].Data))
 
 	// Assert that QueryResultRowsForHost returns empty slice when no results are found
 	results, err = ds.QueryResultRowsForHost(context.Background(), 999, 999)
 	require.NoError(t, err)
 	require.Len(t, results, 0)
 }
 
+func testGetQueryResultRowsForHost(t *testing.T, ds *Datastore) {
+	user := test.NewUser(t, ds, "Test User", "test@example.com", true)
+	query := test.NewQuery(t, ds, nil, "New Query", "SELECT 1", user.ID, true)
+	host1 := test.NewHost(t, ds, "hostname1", "192.168.1.100", "1111", "UI8XB1223", time.Now())
+	host2 := test.NewHost(t, ds, "hostname2", "192.168.1.100", "2222", "UI8XB1223", time.Now())
+
+	mockTime := time.Now().UTC().Truncate(time.Second)
+
+	// Insert 2 Result Rows for Query1 Host1
+	host1ResultRows := []*fleet.ScheduledQueryResultRow{
```

Comment:
Let's also add a test for the common case where a query returns one empty result (just one item with `Data: nil`).